src/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEnumerator.Unix.cs

```diff
@@ -0,0 +1,207 @@
+            FileSystemEntry entry = default;
+
+            Monitor.Enter(_lock, ref acquiredLock);
+
+            try
+            {
+                if (_lastEntryFound)
+                    return false;
+
+                do
+                {
+                    FindNextEntry();
+                    if (_lastEntryFound)
+                        return false;
+
+                    // Get from the dir entry whether the entry is a file or directory.
+                    // We classify everything as a file unless we know it to be a directory.
+                    // (This includes regular files, FIFOs, etc.)
+                    bool isDirectory = false;
+                    if (_entry.InodeType == Interop.Sys.NodeType.DT_DIR)
+                    {
+                        // We know it's a directory.
+                        isDirectory = true;
+                    }
+                    else if (_entry.InodeType == Interop.Sys.NodeType.DT_LNK || _entry.InodeType == Interop.Sys.NodeType.DT_UNKNOWN)
+                    {
+                        // It's a symlink or unknown: stat to it to see if we can resolve it to a directory.
+                        // If we can't (e.g. symlink to a file, broken symlink, etc.), we'll just treat it as a file.
+                        isDirectory = FileSystem.DirectoryExists(Path.Combine(_currentPath, _entry.InodeName));
+                    }
+
+                    if (_options.AttributesToSkip != 0)
+                    {
+                        if (((_options.AttributesToSkip & FileAttributes.Directory) != 0 && isDirectory)
+                            || ((_options.AttributesToSkip & FileAttributes.Hidden) != 0 && _entry.InodeName[0] == '.')
+                            || ((_options.AttributesToSkip & FileAttributes.ReparsePoint) != 0 && _entry.InodeType == Interop.Sys.NodeType.DT_LNK))
+                            continue;
+
+                        // TODO: https://github.com/dotnet/corefx/issues/26715
+                        // Handle readonly skipping
+                    }
+
+                    FileSystemEntry.Initialize(ref entry, _entry, isDirectory, _currentPath, _rootDirectory, _originalRootDirectory);
+
+                    if (isDirectory)
+                    {
+                        // Subdirectory found
+                        if (PathHelpers.IsDotOrDotDot(_entry.InodeName))
+                        {
+                            // "." or "..", don't process unless the option is set
+                            if (!_options.ReturnSpecialDirectories)
+                                continue;
+                        }
+                        else if (_options.RecurseSubdirectories && ShouldRecurseIntoEntry(ref entry))
+                        {
+                            // Recursion is on and the directory was accepted, Queue it
+                            string subDirectory = PathHelpers.CombineNoChecks(_currentPath, _entry.InodeName);
```

Comment:
nit, either the 'd' in subdirectory is capitalized or not. It looks like our pattern is not eg `RecurseSubdirectories` `CreateSubdirectory`

Reply:
fixed- that came with the copy from the old enumerator